fe/fe-core/src/main/java/com/starrocks/load/batchwrite/LoadExecutor.java

```diff
@@ -0,0 +1,276 @@
+        try {
+            Pair<Database, OlapTable> pair = getDbAndTable();
+            timeTrace.buildPlanTimeMs.set(System.currentTimeMillis());
+            LoadPlanner loadPlanner = new LoadPlanner(-1, loadId, txnId, pair.first.getId(),
+                    tableId.getDbName(), pair.second, streamLoadInfo.isStrictMode(), streamLoadInfo.getTimezone(),
+                    streamLoadInfo.isPartialUpdate(), connectContext, null,
+                    streamLoadInfo.getLoadMemLimit(), streamLoadInfo.getExecMemLimit(),
+                    streamLoadInfo.getNegative(), coordinatorBackendIds.size(), streamLoadInfo.getColumnExprDescs(),
+                    streamLoadInfo, label, streamLoadInfo.getTimeout());
+            loadPlanner.setWarehouseId(streamLoadInfo.getWarehouseId());
+            loadPlanner.setBatchWrite(batchWriteIntervalMs, loadParameters, coordinatorBackendIds);
+            loadPlanner.plan();
+
+            coordinator = coordinatorFactory.createStreamLoadScheduler(loadPlanner);
+            QeProcessorImpl.INSTANCE.registerQuery(loadId, coordinator);
+            timeTrace.executePlanTimeMs.set(System.currentTimeMillis());
+            coordinator.exec();
+
+            timeTrace.joinPlanTimeMs.set(System.currentTimeMillis());
+            int waitSecond = streamLoadInfo.getTimeout() -
+                    (int) (System.currentTimeMillis() - timeTrace.createTimeMs.get()) / 1000;
+            if (coordinator.join(waitSecond)) {
+                Status status = coordinator.getExecStatus();
+                if (!status.ok()) {
+                    throw new LoadException(
+                            String.format("Failed to execute load, status code: %s, error message: %s",
+                                    status.getErrorCodeString(), status.getErrorMsg()));
+                }
+                tabletCommitInfo = TabletCommitInfo.fromThrift(coordinator.getCommitInfos());
+                tabletFailInfo = TabletFailInfo.fromThrift(coordinator.getFailInfos());
+            } else {
+                throw new LoadException(
+                        String.format("Timeout to execute load after waiting for %s seconds", waitSecond));
+            }
+        } finally {
+            QeProcessorImpl.INSTANCE.unregisterQuery(loadId);
+        }
+    }
+
+    private Pair<Database, OlapTable> getDbAndTable() throws Exception {
+        GlobalStateMgr globalStateMgr = GlobalStateMgr.getCurrentState();
+        Database db = globalStateMgr.getLocalMetastore().getDb(tableId.getDbName());
+        if (db == null) {
+            throw new LoadException(String.format("Database %s does not exist", tableId.getDbName()));
+        }
+
+        Table table;
+        Locker locker = new Locker();
+        locker.lockDatabase(db.getId(), LockType.READ);
+        try {
+            table = GlobalStateMgr.getCurrentState()
+                    .getLocalMetastore().getTable(db.getFullName(), tableId.getTableName());
```

Comment:
better to use `db.getTable()` and lock is not necessary.

Reply:
done